### PR TITLE
Feature/608 connection config db

### PR DESCRIPTION
**Pull request by HenkKodde**
As per story #608 - Add config to default.json to manage the connection pool for knex and MySQL.

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -13,6 +13,10 @@
     "CONNECT_TIMEOUT": 45000,
     "RECONNECT_INTERVAL": 5000
   },
+  "DB_CONNECTION": {
+    "DB_CONNECTION_POOL_MIN": 10,
+    "DB_CONNECTION_POOL_MAX": 30
```
Comment (mdebarros) [inline config/default.json]: You should remove the `DB_CONNECTION_` prefix portion for `POOL_MAX` as its already a sub-object

File: src/lib/config.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ module.exports = {
   PORT: RC.PORT,
   // ADMIN_PORT: RC.ADMIN_PORT,
   DATABASE_URI: RC.DATABASE_URI,
+  DB_CONNECTION_POOL_MIN: RC.DB_CONNECTION.DB_CONNECTION_POOL_MIN,
```
Comment (mdebarros) [inline src/lib/config.js]: This should be changed from `RC.DB_CONNECTION.DB_CONNECTION_POOL_MIN` to `RC.DB_CONNECTION.POOL_MIN`

File: src/lib/config.js [hunk 3]
```diff
@@ -5,6 +5,8 @@ module.exports = {
...
+  DB_CONNECTION_POOL_MAX: RC.DB_CONNECTION.DB_CONNECTION_POOL_MAX,
```
Comment (mdebarros) [inline src/lib/config.js]: This should be changed from `RC.DB_CONNECTION.DB_CONNECTION_POOL_MAX` to `RC.DB_CONNECTION.POOL_MAX`

File: test/integration-config.json [hunk 4]
```diff
@@ -16,6 +16,10 @@
     "CONNECT_TIMEOUT": 45000,
     "RECONNECT_INTERVAL": 5000
   },
+  "DB_CONNECTION": {
+    "DB_CONNECTION_POOL_MIN": 10,
```
Comment (mdebarros) [inline test/integration-config.json]: Same change above applies here.

File: test/integration-config.json [hunk 5]
```diff
@@ -16,6 +16,10 @@
...
+    "DB_CONNECTION_POOL_MAX": 30
```
Comment (mdebarros) [inline test/integration-config.json]: Same change above applies here.

File: config/default.json [hunk 6]
```diff
@@ -13,6 +13,10 @@
     "CONNECT_TIMEOUT": 45000,
     "RECONNECT_INTERVAL": 5000
   },
+  "DB_CONNECTION": {
+    "DB_CONNECTION_POOL_MIN": 10,
```
[the remark this answers is not in the document]
Comment (HenkKodde) [reply config/default.json]: Makes logical sense - have updated.

File: config/default.json (hunk 1 above)
Comment (HenkKodde) [reply config/default.json]: Makes logical sense - have updated.

File: src/lib/config.js (hunk 2 above)
Comment (HenkKodde) [reply src/lib/config.js]: Makes logical sense - have updated.

File: src/lib/config.js (hunk 3 above)
Comment (HenkKodde) [reply src/lib/config.js]: Makes logical sense - have updated.

File: test/integration-config.json (hunk 4 above)
Comment (HenkKodde) [reply test/integration-config.json]: Updated.

File: test/integration-config.json (hunk 5 above)
Comment (HenkKodde) [reply test/integration-config.json]: Updated.

File: src/lib/config.js [hunk 7]
```diff
@@ -5,6 +5,8 @@ module.exports = {
   PORT: RC.PORT,
   // ADMIN_PORT: RC.ADMIN_PORT,
   DATABASE_URI: RC.DATABASE_URI,
+  POOL_MIN: RC.DB_CONNECTION.POOL_MIN,
```
Comment (mdebarros) [inline src/lib/config.js]: Why was this changed? This was fine. `DB_CONNECTION_POOL_MIN`? It has no context now.

File: src/lib/config.js [hunk 8]
```diff
@@ -5,6 +5,8 @@ module.exports = {
...
+  POOL_MAX: RC.DB_CONNECTION.POOL_MAX,
```
Comment (mdebarros) [inline src/lib/config.js]: Why was this changed? This was fine. `DB_CONNECTION_POOL_MAX`?  It has no context now.

File: test/integration-config.json [hunk 9]
```diff
@@ -16,6 +16,10 @@
     "CONNECT_TIMEOUT": 45000,
     "RECONNECT_INTERVAL": 5000
   },
+  "DB_CONNECTION": {
+    "POOL_MIN": 10,
```
Comment (mdebarros) [inline test/integration-config.json]: Same as above.

File: test/integration-config.json [hunk 10]
```diff
@@ -16,6 +16,10 @@
...
+    "POOL_MAX": 30
```
Comment (mdebarros) [inline test/integration-config.json]: Same as above.

File: src/lib/config.js (hunk 7 above)
Comment (HenkKodde) [reply src/lib/config.js]: Over eager - undo updates.

Comment (HenkKodde) [reply src/lib/config.js]: Revert updates.

File: test/integration-config.json (hunk 9 above)
Comment (HenkKodde) [reply test/integration-config.json]: Revert updates.

File: src/lib/config.js (hunk 8 above)
Comment (HenkKodde) [reply src/lib/config.js]: Revert updates.

File: test/integration-config.json (hunk 10 above)
Comment (HenkKodde) [reply test/integration-config.json]: Have a question regarding this - initial request was to remove the prefix. Can we please verify?

File: package.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mojaloop/central-ledger",
-  "version": "5.5.2",
+  "version": "6.1.0",
```
Comment (mdebarros) [inline package.json]: Please bump up the version to 6.1.1 since there already has been a release.

Comment (HenkKodde) [reply package.json]: Updated release and version to 6.1.1.

**Comment by mdebarros** [comment]
Thanks @HenkKodde
